spec.bs

```diff
@@ -2821,14 +2821,141 @@ a [=list=] of [=interest groups=] |bidIgs|, and a [=list=] of [=bid debug report
+     equals |componentAd|. If there is no matching element, return failure.
+  1. [=list/Append=] |ad| onto |winningAdComponents|.
+1. Let |winningBid| be a new [=generated bid=] with the following [=struct/items=]:
+   : [=generated bid/id=]
+   :: (0, 0)
+   : [=generated bid/for k-anon auction=]
+   :: TODO
+   : [=generated bid/bid=]
+   :: A new [=bid with currency=] with the following [=struct/items=]:
+      : [=bid with currency/value=]
+      :: |response|'s [=server auction response/bid=]
+      : [=bid with currency/currency=]
+      :: |response|'s [=server auction response/bid currency=]
+
+  : [=generated bid/bid in seller currency=]
+  :: TODO
+  : [=generated bid/ad=]
+  :: |response|'s [=server auction response/ad render url=]
+  : [=generated bid/ad descriptor=]
+  :: |winningAd|
+  : [=generated bid/ad component descriptors=]
+  :: |winningAdComponents|
+  : [=generated bid/ad cost=]
+  :: Null
+  : [=generated bid/modeling signals=]
+  :: Null
+  : [=generated bid/interest group=]
+  :: |winningGroup|
+  : [=generated bid/bid ad=]
+  :: |winningAd|
+  : [=generated bid/modified bid=]
+  :: Null
+  : [=generated bid/bid duration=]
+  :: 0
+  : [=generated bid/component seller=]
+  :: Null
+  : [=generated bid/target number of ad components=]
+  :: Null
+  : [=generated bid/number of mandatory ad components=]
+  :: Null
+1. Let |buyerReportingResult| be a new [=reporting result=] with the following [=struct/items=]:
+   : [=reporting result/report url=]
+   :: The value of |response|'s [=server auction response/buyer reporting=]'s [=server auction reporting info/reporting url=].
+   : [=reporting result/reporting beacon map=]
+   :: The value of |response|'s [=server auction response/buyer reporting=]'s [=server auction reporting info/beacon urls=].
+   : [=reporting result/reporting macro map=]
+   :: Null
+
+1. Let |sellerReportingResult| be a new [=reporting result=] with the following [=struct/items=]:
+   : [=reporting result/report url=]
+   :: The value of |response|'s [=server auction response/top level seller reporting=]'s [=server auction reporting info/reporting url=].
+   : [=reporting result/reporting beacon map=]
+   :: The value of |response|'s [=server auction response/top level seller reporting=]'s [=server auction reporting info/beacon urls=].
+   : [=reporting result/reporting macro map=]
+   :: Null
+
+1. Let |componentSellerReportingResult| be a new [=reporting result=] with the following [=struct/items=]:
+   : [=reporting result/report url=]
+   :: The value of |response|'s [=server auction response/component seller reporting=]'s [=server auction reporting info/reporting url=].
+   :: |response.winReportingURLs.componentSellerReportingUrls.reportingUrl|
+   : [=reporting result/reporting beacon map=]
+   :: The value of |response|'s [=server auction response/component seller reporting=]'s [=server auction reporting info/beacon urls=].
+   : [=reporting result/reporting macro map=]
+   :: Null

```

Comment:
this has a default value null (https://wicg.github.io/turtledove/#reporting-result-reporting-macro-map), so if it's null, just no need to set it.

Reply:
Done.